internal/test/commands/test_commands.go

```diff
@@ -355,28 +355,52 @@ func (self *TestCommands) LineageTable() datatable.DataTable {
 
 // LocalBranches provides the names of all branches in the local repository,
 // ordered alphabetically.
-func (self *TestCommands) LocalBranches() (gitdomain.LocalBranchNames, error) {
-	output, err := self.QueryTrim("git", "branch")
+func (self *TestCommands) LocalBranches() (allBranches, branchesInOtherWorktrees gitdomain.LocalBranchNames, err error) {
+	forEachRefFormat := strings.Join(
+		[]string{
+			"%(if)", "%(HEAD)", "%(then)", // If the branch is checked out in the current worktree
+			"H", // literal "H"
+			"%(else)",
+			"%(if)", "%(worktreepath)", "%(then)", // If the branch is checked out in any (other) worktree
+			"W", // literal "W"
+			"%(else)",
+			" ", // literal " "
+			"%(end)",
+			"%(end)",
+
+			"%(refname:lstrip=2)", // the branch name (without refs/heads/)
+		},
+		"")
+	output, err := self.Query("git", "for-each-ref", "--format="+forEachRefFormat, "refs/heads/")
 	if err != nil {
-		return gitdomain.LocalBranchNames{}, err
+		return gitdomain.LocalBranchNames{}, gitdomain.LocalBranchNames{}, err
 	}
-	result := gitdomain.LocalBranchNames{}
 	for _, line := range stringslice.Lines(output) {
-		line = strings.Trim(line, "* ")
-		line = strings.TrimSpace(line)
-		result = append(result, gitdomain.NewLocalBranchName(line))
+		marker := line[0]
+		branch := line[1:]
+		switch marker {
+		case 'H':
+			fallthrough
+		case ' ':
+			allBranches = append(allBranches, gitdomain.NewLocalBranchName(branch))
```

Comment:
You can group multiple case statements together like this:

```suggestion
		case 'H', ' ':
			allBranches = append(allBranches, gitdomain.NewLocalBranchName(branch))
```